TS_HW1/home_work_1.ts

```diff
@@ -1,112 +1,121 @@
 class School {
-  directions = [];
+  public directions: any[] = []; // має бути : Direction[]
 
-  addDirection(direction) {
+  public addDirection(direction: any): void {
     this.directions.push(direction);
   }
 }
 
 class Direction {
-  levels = [];
+  public levels: any[] = []; // : Level[]
+  private _name: string;
 
-  get name() {
+  get name(): string {
     return this._name;
   }
 
-  constructor(name) {
+  constructor(name: string) {
     this._name = name;
   }
 
-  addLevel(level) {
+  public addLevel(level: any): void {
     this.levels.push(level);
   }
 }
 
 class Level {
-  groups = [];
+  public groups: any[] = []; // : Group[]
+  private _name: string;
+  private _program: string;
 
-  constructor(name, program) {
-    this.name = name;
+  constructor(name: string, program: string) {
+    this._name = name;
     this._program = program;
   }
 
-  get name() {
+  get name(): string {
     return this._name;
   }
 
-  get program() {
+  get program(): string {
     return this._program;
   }
 
-  addGroup(group) {
+  public addGroup(group: any) {
     this.groups.push(group);
   }
 }
 
 class Group {
-  _students = [];
-
-  get students() {
+  private _students: any[] = []; // : Student[]
+  public directionName: string;
+  public levelName: string;
+  
+  get students(): any[] {
     return this._students;
   }
 
-  constructor(directionName, levelName) {
+  constructor(directionName: string, levelName: string) {
     this.directionName = directionName;
     this.levelName = levelName;
   }
 
-  addStudent(student) {
+  public addStudent(student: any) {
```

Comment:
Please, do not forget explicitly add '_void_' as a return type for function:
`public addStudent(student: any): void {}`

Reply:
yes, that's my inattention, here and after